### ISSUE TITLE
macOS - Code unusable: "The window has crashed (reason: 'crashed', code: '5')" (2023-02, v1.75.1)

### ISSUE THREAD

**Issue by philippefutureboy** [microsoft/vscode#175581]
Related to: #132792

Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.75.1 (February 8, 2023 at 4:47PM)
- OS Version: macOS Monterey 12.6.3
- Machine specs:
  - MacBook Pro (16-inch, 2019)
  - Processor: 2.4GHz 8-Core Intel Core i9
  - Graphics: AMD Radeon Pro 5500M 8 GB

**Steps to Reproduce:**

1. Kill any outstanding Code-related processes using Activity Monitor
2. Clean uninstall VSCode (following this answer: https://stackoverflow.com/a/47561948/4442749)
3. Restart computer
4. Download VSCode universal darwin app from https://code.visualstudio.com/download
5. Unzip installer; move VSCode universal app to the Application folder
6. Double click the VSCode icon
7. Cry:

![Screen Shot 2023-02-27 at 3 39 56 PM](https://user-images.githubusercontent.com/31741972/221682998-4384032c-5268-46ef-97ba-c9327a92a0e8.png)

**Attempted steps:**

- Clean reinstall (as per above steps).

- Attempted to generate crash report as per comment on initial issue: https://github.com/microsoft/vscode/issues/132792#issuecomment-918988097 -> Creates a directory of empty folders with a settings.dat file toplevel of size ~152Bytes

---
Can you link me to a few versions previous to 1.75.1? I can test them to see if I can get one to work

**Comment by rzhao271** [comment]
I'm unable to reproduce the issue.
Previous versions to test can be found on release notes pages, such as at https://code.visualstudio.com/updates/v1_74.
If the program continues to crash, could you continue trying https://github.com/microsoft/vscode/wiki/Native-Crash-Issues to see if a dmp file is generated? You might have to wait for `code` to crash, and then press `Close`, before searching for the file.

**Comment by philippefutureboy** [comment]
Hi @rzhao271! Thanks for the quick answer 😃 

## Version 1.74.1

- Version 1.74.1 works following a clean install (minus the restart of the computer), but on second boot crashes similarly.
- Compared to v1.75.1, it loads the sidebar background before failing:

![image](https://user-images.githubusercontent.com/31741972/221725162-3a396b13-3013-4cbe-8a64-04b62cc0ae35.png)

- Version 1.74.1 does not produce a .dmp file when run via `code --crash-reporter-directory ~/some/directory`

## Version 1.73.1

- Version 1.73.1 works following an unclean install (simply delete universal app, download, move to application folder, execute).
- Version 1.73.1 seems stable after trying multiple quit-start cycles via executing the universal app.
- I am able to load my workspace.
- I was able to install several extensions (python, jupyter, terraform, docker, eslint, es6 snippets, react es7, dbt-formatter, dbt power user) and everything worked fine.
- I was able to quit & restart vscode with extensions without issue

But then, I attempted running `code --crash-reporter-directory ~/some/directory` and vscode crashed with the same error. Further restarts via either `code` or the universal executable result in the same crash loop.
Good news is, I have dmp files for the first and second runs. Here is the gist:
https://gist.github.com/philippefutureboy/f695c8f3a703bd4bb27497b896613efe

The files names are prefixed with `1__` and `2__` to symbolize the crash order.


Progress :D

**Comment by philippefutureboy** [comment]
Additional context that may be relevant (or just too much info):

- This issue arose at the same time as:
  1. My Spotify install stopped being able to stream to my bluetooth earbuds; A clean reinstall of Spotify (delete app executable, delete app files, restart, download, reinstall, reexecute) did the trick;
  2. YouTube landing page (wall of videos) started not acknowledging my clicks (page becomes essentially non-interactive, but scrollable)
- I have had recurrent crashes of Chrome (recurrent=once a month) for as long as I have had this computer (2 years now) where all of my browser instances (three different Chrome profiles) crash all at the same time (the window becomes full red (#ff0000) and exits). Originally it was once every three days but after a while it slowed down.
- I use my Macbook Pro in closed-lid mode, with two HDMI output screens. About every second boot result in momentous visual glitch & freeze of the lock screen for 2-5 seconds.

TL;DR: My GPU is not always stable, and that may play a part in this. 

---

No recent installations or manual upgrade of dependencies. Some python deps installed in a virtual env.

**Comment by rzhao271** [comment]
Thanks for the info! I hope to symbolicate the crash dumps later today, but I'm also wondering whether running `code --disable-gpu` works better?

**Comment by rzhao271** [comment]
I just noticed the gists actually have the logs. I'm wondering if you could upload the dmp files instead? I have access to some more symbols, so I can get more detailed logs going.
Also, can you remind me what Electron version 1.73.1 uses? You can find it by going to Help > About.

**Comment by philippefutureboy** [comment]
### RE progress:

I played around with the different conditions:

- Variable 1: Execute from Application folder (via mouse-click) --vs-- execute via `code`
- Variable 2: Kill VSCode via the dock (right-click + Quit) between test runs --vs-- not kill between runs
- Variable 3: Run with `--disable-extensions`
- Variable 4: Run with `--disable-gpu`
- Variable 5: Run with `--crash-reporter-directory`
- Variable 6: Install extensions
- Variable 7: Load a workspace

**And so my primary observation is that VSCode crashes only once a workspace (repository) has been loaded, exited, then restarted.**

Any other variable seems to be independent from the crash.

Any idea why that would be the case?
Any steps I can take to gain further information?
If I can remove the state that says which workspace to load at start, I can test what happens post-crash if the workspace is not opened at launch.

### RE dumps:

It seems like the Gists don't support binary files, so I uploaded the dumps and logs to the following Google Drive directory:
https://drive.google.com/drive/folders/1WeUVsM4FO5I8e9TLPnQxd6e37mkQ_DZ0?usp=sharing

### RE `code --disable-gpu`:

- Running on an already broken v1.73.1 results in same crash;
- Running on a pseudo-clean install (no restart) does not seem to affect negatively the application.
- 

Side note: v.1.73.1 has kept all of the extensions & remembers recent workspaces through reinstall even though I ran the following:

```
rm -rf "~/.vscode"
rm -rf "~/Library/Saved Application State/com.microsoft.VSCode.savedState"
rm -rf "~/Library/Preferences/com.microsoft.VSCode.plist"
rm -rf  "~/Library/Preferences/com.microsoft.VSCode.helper.plist"
rm -rf "~/Library/Caches/com.microsoft.VSCode"
rm -rf "~/Library/Application Support/com.apple.sharedfilelist/com.apple.LSSharedFileList.ApplicationRecentDocuments/com.microsoft.vscode.sfl*"
rm -rf "~/Library/Application Support/Code"
```

Are there other folders or files I should delete when doing a clean reinstall?
If so let me know and I'll update the stackoverflow answer on how to uninstall vscode.

### RE v1.73.1 - electron version

Version: 1.73.1 (Universal)
Commit: 6261075646f055b99068d3688932416f2346dd3b
Date: 2022-11-09T02:08:38.961Z (3 mos ago)
Electron: 19.0.17
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Darwin x64 21.6.0
Sandboxed: No

**Comment by rzhao271** [comment]
Thanks for the info!

Symbolicated traces:
[first-symbolicated.log](https://github.com/microsoft/vscode/files/10855679/first-symbolicated.log)
[second-symbolicated.log](https://github.com/microsoft/vscode/files/10855680/second-symbolicated.log)

Something interesting that showed up is the dmp files actually required symbols from Electron v19.1.9 to symbolicate, instead of v19.0.17.

One way to try reproducing the issue with a fresh environment is to use a custom user-data-dir like so:
```sh
mkdir freshu
mkdir freshe
code --user-data-dir=freshu --extensions-data-dir=freshe
```

I don't think you need a custom extensions data dir if you're already running with extensions disabled, but I added it in just in case.

**Comment by deepak1556** [comment]
> Something interesting that showed up is the dmp files actually required symbols from Electron v19.1.9 to symbolicate, instead of v19.0.17.

Updates were not disabled after installing `1.73` so application was updated to `1.75` before the crash happened.

Looks like a crash during deserialization of code cache, @philippefutureboy does the crash repro when you launch with `--no-cached-data` ?

**Comment by philippefutureboy** [comment]
### RE: `code --no-cached-data`

Running with `--no-cached-data` results in no visible changes, except following warning:

`Warning: 'cached-data' is not in the list of known options, but still passed to Electron/Chromium.`

### RE: Automated updates

As @deepak1556 pointed out, v1.73.1 seems to update automatically to v1.75.1 after a few minutes.
To confirm this, I ran `code --version && code {{args}}` for each test, and I confirm that v1.75.1 auto installs and then code starts failing.

An interesting information is that I need to quit the app in the dock before the crash starts.
So, anything done within v1.73.1 is fine; v1.75.1 results in app crashing only following quitting the app in the dock and restarting. I hypothesize that some data must corrupt between the first v1.75.1 run and the second run.

### Additional info - list of extensions installed

```
code --list-extensions
4ops.terraform
bastienboutonnet.vscode-dbt
dbaeumer.vscode-eslint
dsznajder.es7-react-js-snippets
henriblancke.vscode-dbt-formatter
ms-azuretools.vscode-docker
ms-python.isort
ms-python.python
ms-python.vscode-pylance
ms-toolsai.jupyter
ms-toolsai.jupyter-keymap
ms-toolsai.jupyter-renderers
ms-toolsai.vscode-jupyter-cell-tags
ms-toolsai.vscode-jupyter-slideshow
samuelcolvin.jinjahtml
xabikos.JavaScriptSnippets
```

**Comment by rzhao271** [comment]
I don't think the issue has to do with the extensions if it still occurs even when the extensions are disabled. I'm wondering if the issue is fixed when using the [Exploration build](https://az764295.vo.msecnd.net/exploration/e4a11014fd91da8f2d82ba144097df21a483334d/VSCode-darwin-universal.zip)? It uses a newer version of Electron.

**Comment by philippefutureboy** [comment]
Just to keep the thread fresh - I'll be trying the exploration build by Tue evening (EST time)!
For now I fixed my vscode version to 1.73.1 and downgraded extension versions to matching versions.

**Comment by philippefutureboy** [comment]
@rzhao271 
Sorry for the delay in my answer - life got in the way.

**I confirm that the Exploration build seems stable.**

So for now I guess that my best choice is to stick with 1.73.1 (disable updates and extension updates) and wait for the current version of the Exploration build to merge in the main distribution?

**Comment by philippefutureboy** [comment]
Marking as closed, since I have a stable version running (1.73.1) and will be able to migrate to the exploration build in due time.
